File: wrappers/csharp/Intel.RealSense/Types/Extrinsics.cs
// License: Apache 2.0. See LICENSE file in root directory.
// Copyright(c) 2017 Intel Corporation. All Rights Reserved.

namespace Intel.RealSense
{
    using System;
    using System.Collections.Generic;
    using System.Runtime.InteropServices;

    /// <summary>
    /// Cross-stream extrinsics: encode the topology describing how the different devices are connected.
    /// </summary>
    [System.Serializable]
    [StructLayout(LayoutKind.Sequential)]
    public struct Extrinsics
    {
        /// <summary>
        /// Column-major 3x3 rotation matrix
        /// </summary>
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 9)]
        public float[] rotation;

        /// <summary>
        /// Three-element translation vector, in meters
        /// </summary>
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 3)]
        public float[] translation;
    }
}

File: wrappers/android/librealsense/src/main/java/com/intel/realsense/librealsense/Extrinsics.java
package com.intel.realsense.librealsense;

public class Extrinsics {

    private float[] mRotation; // Column-major 3x3 rotation matrix
    private float[] mTranslation;  // Three-element translation vector, in meters

    public Extrinsics(){
        mRotation = new float[9];
        mTranslation = new float[3];
    }

    public Extrinsics(float[] rotation, float[] translation){
        this.mRotation = rotation;
        this.mTranslation = translation;
    }
}
